### PR TITLE
Theme script

### PR DESCRIPTION
**Pull request by mzohaibqc**
This PR is about replacing `scripts/generateColorLess.js` with `antd-theme-generator` which allows to change all color related variables and change theme dynamically in browser.
Here is an demo: https://antd-live-theme.firebaseapp.com/
Github: https://github.com/mzohaibqc/antd-theme-generator
issue: #10384 

themedemo

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x ] Rebase before creating a PR to keep commit history clear.
* [x ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

**Comment by mzohaibqc** [comment]
@yesmeck @afc163 Kindly have a look

File: scripts/generateColorLess.js [hunk 1]
```diff
@@ -1,92 +1,16 @@
...
-  '#1890ff': '@primary-color',
-  '#096dd9': 'color(~`colorPalette("@{primary-color}", 7)`)', // @primary-7
-  '#46a6ff': 'tint(@primary-color, 20%)',
-  '#8cc8ff': 'tint(@primary-color, 50%)',
-  'rgba\\(24, 144, 255, 0.2\\)': 'fade(@primary-color, 20%)',
+const { generateTheme } = require('antd-theme-generator');
+
+const options = {
+  stylesDir: path.join(__dirname, '../site/theme/static'),
+  antDir: path.join(__dirname, '../node_modules/antd'),
```
Comment (yesmeck) [inline scripts/generateColorLess.js]: Can we use `path.join(__dirname, '../components')` ?

File: scripts/generateColorLess.js [hunk 2]
```diff
@@ -1,92 +1,16 @@
...
+  varFile: path.join(__dirname, '../node_modules/antd/lib/style/themes/default.less'),
```
Comment (yesmeck) [inline scripts/generateColorLess.js]: Same here, can we use `path.join(__dirname, '../components/style/themes/default.less')`

**Comment by mzohaibqc** [comment]
@yesmeck I added your feedback changes.

**Comment by yesmeck** [comment]
@mzohaibqc Build faild

**Comment by mzohaibqc** [comment]
@yesmeck After changing `antd` path, now getting this error on build, Any idea?

```Can't resolve '../default.deperated' in '/opt/build/repo/components/style/themes'
9:37:06 AM:       in /opt/build/repo/components/style/mixins/reset.less (line null, column -1)
9:37:06 AM:     at runLoaders (/opt/build/repo/node_modules/webpack/lib/NormalModule.js:195:19)
9:37:06 AM:     at /opt/build/repo/node_modules/loader-runner/lib/LoaderRunner.js:364:11
9:37:06 AM:     at /opt/build/repo/node_modules/loader-runner/lib/LoaderRunner.js:230:18
9:37:06 AM:     at context.callback (/opt/build/repo/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
9:37:06 AM:     at <anonymous>
9:37:06 AM:  @ ./components/tabs/style/index.tsx 5:0-23
9:37:06 AM:  @ ./components ^\.\/[^_][\w-]+\/style\/index\.tsx?$
9:37:06 AM:  @ ./index.js
9:37:06 AM:  @ ./site/theme/template/Content/Article.jsx
9:37:06 AM:  @ ./site/theme ^\.\/template.*$
9:37:06 AM:  @ ./node_modules/bisheng/tmp/routes.index.js
9:37:06 AM:  @ ./node_modules/bisheng/tmp/entry.index.js
9:37:06 AM: ERROR in ./components/button/style/index.less
9:37:06 AM: Module build failed: ModuleBuildError: Module build failed:```

**Comment by mzohaibqc** [comment]
@yesmeck  Using less files from `components` directory is working fine locally but build is failing on `travis-ci`. May I change paths back to antd directory from `node_modules`? Becasue I don't understand why is this happening. Let me know if you have any idea why this is failing.

**Comment by yesmeck** [comment]
The error has been fixed on master, could you rebase master?

**Comment by mzohaibqc** [comment]
@yesmeck I rebased master in my branch and now at's all green :) Thanks

Kindly merge

**Comment by mzohaibqc** [comment]
@yesmeck Also let me know what else I can do for this project. I've tight schedule but I'll try

**Comment by yesmeck** [comment]
Thanks, you do a great job!

**Pull request merged**